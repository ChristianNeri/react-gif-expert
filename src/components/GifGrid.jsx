import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs( category );

    // const [counter, setCounter] = useState(10);




    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && ( <h2>Cargando...</h2> )
                // ? ( <h2>Cargando...</h2> )
                // : null
            }
            

            <div className="card-grid">
                {/* images.map */}
                {
                    images.map( (image)  => (
                        <GifItem 
                        key={ image.id }
                        { ...image }
                        // title = { image.title }
                        // url = { image.url }
                        />
                    ))
                }
            </div>
            {/* <h5>{ counter }</h5>
            <button  onClick={ () => setCounter( counter + 1 ) }>+1</button> */}
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}