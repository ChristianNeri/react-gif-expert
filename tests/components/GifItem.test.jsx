import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Pruebas a GifItem.jsx', () => { 

    const title = 'Demon Slayer';
    const url = 'https://google.com/';
    test('debe de hacer match con el snapshot', () => { 
        
        const { contaier } = render( <GifItem title={title} url={ url }/> );
        expect( contaier ).toMatchSnapshot();
     })

     test(' Debe mostrar la imagen con el URL y el ALT indicado', () => { 
        
        render( <GifItem title={ title } url={ url } /> )
        // expect( screen.getByRole('img').src ).toBe(url)
        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );

      })

      test('Debe de mostrar el titulo en el componente', () => { 
        render( <GifItem title={ title } url={ url } /> )
        expect( screen.getByText( title ) ).toBeTruthy();

       })
 })