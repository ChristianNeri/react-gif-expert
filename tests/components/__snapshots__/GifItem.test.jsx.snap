// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GifItem.jsx debe de hacer match con el snapshot 1`] = `undefined`;
